Fix docs and possibly code to support running functional tests on macOS

We found a few [documentation](https://github.com/EFForg/privacybadger/blob/e97b0d0dbe88ea8a08f9f4b5b06f31807bce741d/doc/tests.md#functional-tests) nits and what may be platform-specific code preventing tests from running on macOS.

Documentation nits:
- gecko/chromedriver installation instructions should point out you need different binaries for macOS.
- The browser binary doesn't seem to be in your `$PATH` on macOS; so `BROWSER=firefox` doesn't work. Should give an example of running tests with a typical macOS Firefox binary path.

cc @alexristich

A quick note on this: when running on MacOS, you need to provide the path to the executable, e.g. `BROWSER=/Applications/Firefox.app/Contents/MacOS/firefox pytest -v`.

However, when doing this for both Firefox and Chrome I receive errors. 

For Chrome it shows:
```
__________________________________ ERROR collecting tests/selenium/pbtest_org_test.py __________________________________
/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages/_pytest/python.py:209: in fget
    return self._obj
E   AttributeError: 'Module' object has no attribute '_obj'

During handling of the above exception, another exception occurred:
tests/selenium/pbtest_org_test.py:4: in <module>
    import pbtest
tests/selenium/pbtest.py:185: in <module>
    shim = Shim()  # create the browser shim
tests/selenium/pbtest.py:96: in __init__
    self.browser_type = get_browser_type(self.browser_path)
tests/selenium/pbtest.py:48: in get_browser_type
    raise ValueError("couldn't get browser type from %s" % string)
E   ValueError: couldn't get browser type from /Applications/Google Chrome.app/Contents/MacOS/Google Chrome
```

For Firefox it shows:
```
tests/selenium/pbtest.py:274: in run
    with self.manager() as driver:
/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/contextlib.py:59: in __enter__
    return next(self.gen)
tests/selenium/pbtest.py:175: in firefox_manager
    driver = webdriver.Firefox(firefox_profile=ffp, firefox_binary=self.browser_path)
/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages/selenium/webdriver/firefox/webdriver.py:80: in __init__
    self.binary, timeout)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <selenium.webdriver.firefox.extension_connection.ExtensionConnection object at 0x105019e80>, host = '127.0.0.1'
firefox_profile = <selenium.webdriver.firefox.firefox_profile.FirefoxProfile object at 0x105019198>
firefox_binary = '/Applications/Firefox.app/Contents/MacOS/firefox', timeout = 30

    def __init__(self, host, firefox_profile, firefox_binary=None, timeout=30):
        self.profile = firefox_profile
        self.binary = firefox_binary
        HOST = host
        timeout = int(timeout)
    
        if self.binary is None:
            self.binary = FirefoxBinary()
    
        if HOST is None:
            HOST = "127.0.0.1"
    
        PORT = utils.free_port()
        self.profile.port = PORT
        self.profile.update_preferences()
    
        self.profile.add_extension()
    
>       self.binary.launch_browser(self.profile, timeout=timeout)
E       AttributeError: 'str' object has no attribute 'launch_browser'

/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages/selenium/webdriver/firefox/extension_connection.py:52: AttributeError
```

--- a/.gitignore
+++ b/.gitignore
@@ -43,6 +43,7 @@ node_modules
 
 tests/.cache
 tests/.pytest_cache
+.pytest_cache
 
 *.xpi
 .chrome-profile

--- a/doc/tests.md
+++ b/doc/tests.md
@@ -47,6 +47,9 @@ The `BROWSER` environment variable must be set. It must be one of:
 * `BROWSER=/path/to/a/browser`
 * the name of a browser executable that can be found like `which $BROWSER`
 * or simply `BROWSER=chrome` or `BROWSER=firefox` if you have them installed
+* For MacOS and Firefox, the full Application folder path to the binary is:
+  `BROWSER=/Applications/Firefox.app/Contents/MacOS/firefox-bin`
+* For MacOS and Chrome (complete with spaces that need to be escaped) use: `BROWSER=/Applications/Google\ Chrome.app/Contents/MacOS/Google\ Chrome`
 
 ### Examples
 

--- a/scripts/setup_travis.sh
+++ b/scripts/setup_travis.sh
@@ -33,8 +33,17 @@ function setup_firefox {
         exit 1
       fi
     fi
+
+    # Geckodriver distribution is MacOS or Linux specific
+    os="$(uname -s)"
+    if [[ $os == "Darwin" ]]; then
+      os_dist="macos.tar.gz"
+    else
+      os_dist="linux64.tar.gz"
+    fi
+
     echo "Setting up geckodriver version $version ..."
-    url="https://github.com/mozilla/geckodriver/releases/download/${version}/geckodriver-${version}-linux64.tar.gz"
+    url="https://github.com/mozilla/geckodriver/releases/download/${version}/geckodriver-${version}-${os_dist}"
     wget -q -O /tmp/geckodriver.tar.gz "$url"
     sudo tar -xvf /tmp/geckodriver.tar.gz -C /usr/local/bin/
     sudo chmod a+x /usr/local/bin/geckodriver

